Do not raise exceptions in teardown

flub opened the pull request:
This avoids double-reporting of failures and formats the failure
a bit nicer.

Any exception in teardown is an error, not a test failure.  Instead
modify the runtest protocol call step to perform the checks that the
fixture wants to do at the end of the tests.

The marker system is a great way to transfer state on the test item
since it uses a plublic API to both add the information as well as
access it.  The marker itself is not specifically tied to this
fixture.

This also modifies the default run to include a report of failed
tests.

Commit:
Do not raise exceptions in teardown

Any exception in teardown is an error, not a test failure.  Instead
modify the runtest protocol call step to perform the checks that the
fixture wants to do at the end of the tests.

The marker system is a great way to transfer state on the test item
since it uses a plublic API to both add the information as well as
access it.  The marker itself is not specifically tied to this
fixture.

This also modifies the default run to include a report of failed
tests.

Commit:
Add newline at end of file

jan-auer commented on pytest.ini:
```diff
@@ -1,2 +1,7 @@
 [pytest]
-addopts = --tb=native --durations 5
+addopts = --tb=native --durations 5 -ra
```
Cool, this means we can remove this from `.travis.yml`.

jan-auer commented on tests/integration/fixtures/mini_sentry.py:
```diff
@@ -105,12 +105,14 @@ def check_challenge():
-                AssertionError(
-                    "Relay sent us event: %s" % get_error_message(flask_request.data)
-                ),
+                AssertionError("Relay sent us event:\n{}".format(pformat(data))),
```
Is there a problem with `get_error_message`? This was specifically introduced to render readable errors rather than endless JSON that contains no useful information. Same goes for `format_failures()` below.

Also, this means that `get_error_message` is dead code now.

jan-auer replied:
You can see an example of that in this very build actually:
https://travis-ci.com/github/getsentry/relay/jobs/361775792#L1549

The intention is to *only* print the error message(s), which is extracted by `get_error_message` from the event payload, and then formatted by `format_failures()`.

flub replied:
oh sorry :(. Thanks for catching.  I was working from a version where `get_error_message()` didn't exist yet.  I'll put it back.

jan-auer reviewed (changes requested):
Deferring final review to @untitaker

untitaker commented on tests/integration/fixtures/mini_sentry.py:
```diff
@@ -191,15 +193,24 @@ def outcomes():
+
         if sentry.test_failures:
-            raise AssertionError(
-                f"Exceptions happened in mini_sentry: {sentry.format_failures()}"
```
Also format_failures is now probably dead code?

untitaker reviewed (approved):
This seems fine, I am a bit unsure about the additional formatting changes and the resulting dead code.

Commit:
Keep using get_error_message()

Got removed by accident.

Commit:
Remove pytest -ra flag from makefile

This is now in pytest.ini

Commit:
Use format_failures again

This would have really helped if I pulled recent changes before making
that PR...

Commit:
Typo

Commit:
Spelling and grammar

Commit:
Revert accidental change

jan-auer commented on tests/integration/fixtures/mini_sentry.py:
```diff
@@ -194,12 +195,19 @@ def fail(e):
 
-    @request.addfinalizer
     def reraise_test_failures():
+        from pprint import pprint
```
This import is no longer needed.

jan-auer commented on tests/integration/fixtures/mini_sentry.py:
```diff
@@ -194,12 +195,19 @@ def fail(e):
+            msg = "{n} exceptions happened in mini_sentry:\n\n".format(
+                n=len(sentry.test_failures)
             )
+            msg += sentry.format_failures()
```
_nit_: Can this be added to the above `format()`?

jan-auer reviewed (approved):
Thanks, these double-failures have been super confusing!

Commit:
Cleanup string formatting a little more
